src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx

```diff
@@ -68,8 +68,8 @@ const ReimbursementProductTable: React.FC<ReimbursementProductTableProps> = ({
     }
   });
 
-  const onCostBlurHandler = (value: number, name: string) => {
-    setValue(name, value.toFixed(2));
+  const onCostBlurHandler = (value: number, name: `reimbursementProducts.${number}.cost`) => {
```

Comment:
why you passing in name like this? If you wanna pass in the value directly instead of typing it, you should remove name as a parameter and replace where it's called with `reimbursementProducts.${number}.cost`